manifests/plugin/host_reports.pp

```diff
@@ -0,0 +1,58 @@
+# = Foreman Proxy Host Reports plugin
+#
+# This class installs the Host Reports plugin
+#
+# === Parameters:
+#
+# $proxy_name::               Proxy hostname to appear in reports JSON
+#
+# $spool_dir::                Spool directory with processed reports
+#
+# $keep_reports::             Keep sent reports in spool_dir directory
+#                             when enabled, move files from the place on
+#                             a regular basis (e.g. via cronjob).
+#
+# $report_type::              Set to "proxy" to actually enable the new format of reports,
+#                             set to "foreman" for no changes.
+#
+# === Advanced parameters:
+#
+# $enabled::                  enables/disables the host_reports plugin
+#
+# $listen_on::                proxy feature listens on http, https, or both
+#
+# $version::                  plugin package version, it's passed to ensure parameter of package resource
+#                             can be set to specific version number, 'latest', 'present' etc.
+#
+class foreman_proxy::plugin::host_reports (
+  Optional[String] $proxy_name = undef,
+  Stdlib::Absolutepath $spool_dir = '/var/lib/foreman-proxy/host_reports',
+  Optional[String] $keep_reports = false,
+  # No reports are redirected by default
+  Optional[String] $report_type = 'foreman',
```

Comment:
If there's a default, it's not `Optional`. In Puppet the type `Optional[String]` is equivalent to `Variant[Undef, String]` but you can't get back to `undef` if there's a default. If there are just 2 values possible (as the documentation suggests) you can also use an even stricter type:
```suggestion
  Enum['foreman', 'proxy'] $report_type = 'foreman',
```